projects/gloo/pkg/translator/listener.go

```diff
@@ -63,6 +65,24 @@ func (t *translatorInstance) computeListener(params plugins.Params, proxy *v1.Pr
 			filterChains = append(filterChains, result...)
 		}
 	}
+	// Check for overlapping filterChains according to the same rules that envoy uses here:
+	// https://github.com/envoyproxy/envoy/blob/v1.15.0/source/server/filter_chain_manager_impl.cc#L162-L166
+	for idx1, filterChain := range filterChains {
+		for idx2, otherFilterChain := range filterChains {
+			// only need to compare each pair once
+			if idx2 <= idx1 {
+				continue
+			}
+			if reflect.DeepEqual(filterChain.FilterChainMatch, otherFilterChain.FilterChainMatch) {
+				logger.Debugf("Tried to apply multiple filter chains "+
+					"with the same FilterChainMatch value: {%s}", filterChain.FilterChainMatch.String())
+				validation.AppendListenerError(listenerReport,
+					validationapi.ListenerReport_Error_SSLConfigError, fmt.Sprintf("Tried to apply multiple filter chains "+
+						"with the same FilterChainMatch value: {%s}", filterChain.FilterChainMatch.String()))
```

Comment:
a shorthand way to accomplish the same thing is to use a `%v` formatting print, i.e.
```suggestion
						"with the same FilterChainMatch value: %v", filterChain.FilterChainMatch))
```